service/src/main/java/dk/kvalitetsit/stakit/service/MailQueueServiceImpl.java

```diff
@@ -66,20 +67,22 @@ public void queueStatusUpdatedMail(long serviceConfigurationId, long serviceStat
 
     @Override
     @Transactional
-    public void queueAnnouncementMail(long announcementId) {
+    public void queueAnnouncementMail() {
         var mails = mailSubscriptionDao.findAnnouncementSubscriptions();
 
         if(mails.isEmpty()) {
             return;
         }
 
-        var announcement =announcementDao.getById(announcementId);
+        var announcements = announcementDao.getAnnouncementsToSend();
 
-        mails.stream()
-                .map(x -> new MessageModel(x.email(),
-                        "Stakit Announcement",
-                        announcement.get().message() + "\n\n" + baseUrl + "/unsubscribe/" + x.uuid()))
-                .forEach(this::processMail);
+        for (AnnouncementEntity announcement: announcements) {mails.stream()
+                    .map(x -> new MessageModel(x.email(),
+                            "Stakit Announcement",
+                            announcement.message() + "\n\n" + baseUrl + "/unsubscribe/" + x.uuid()))
+                    .forEach(this::processMail);
+            announcementDao.updateAnnouncementToSent(announcement);
+        }
```

Comment:
Jeg tror følgende scenarie giver problemer. 
1. I en kørsel er der to announcements der skal sendes. 
2. I forbindelse med afsendelse af den anden går der noget galt og der kastes en exception.
3. At der kastes en exception betyder at der bliver lavet et rollback i databasen da metoden er `@Transactional. `
4. Næste gang metoden kaldes vil begge announcements blive sendt igen da opdateringen i linie 84 er rullet tilbage i forbindelse med rollback i step 3.